apps/bmd/src/AppRoot.tsx

```diff
@@ -141,898 +147,1091 @@ export const activationStorageKey = 'activation'
+      return (
+        <PrintOnlyScreen
+          ballotStyleId={ballotStyleId}
+          ballotsPrintedCount={ballotsPrintedCount}
+          electionDefinition={optionalElectionDefinition}
+          isLiveMode={isLiveMode}
+          isVoterCardPresent={isVoterCardPresent}
+          markVoterCardPrinted={markVoterCardPrinted}
+          precinctId={precinctId}
+          printer={printer}
+          useEffectToggleLargeDisplay={useEffectToggleLargeDisplay}
+          showNoChargerAttachedWarning={!hasChargerAttached}
+          updateTally={updateTally}
+          votes={votes}
+        />
+      )
+    }
+    if (isPollsOpen && appMode.isVxMark) {
+      if (isVoterCardPresent && ballotStyleId && precinctId) {
+        if (appPrecinctId !== precinctId) {
           return (
-            <Gamepad onButtonDown={handleGamepadButtonDown}>
-              <BallotContext.Provider
-                value={{
-                  machineConfig,
-                  ballotStyleId,
-                  contests,
-                  electionDefinition,
-                  updateTally: this.updateTally,
-                  isLiveMode,
-                  markVoterCardPrinted: this.markVoterCardPrinted,
-                  markVoterCardVoided: this.markVoterCardVoided,
-                  precinctId,
-                  printer: this.props.printer,
-                  resetBallot: this.resetBallot,
-                  setUserSettings: this.setUserSettings,
-                  updateVote: this.updateVote,
-                  forceSaveVote: this.forceSaveVote,
-                  userSettings,
-                  votes: votes ?? blankBallotVotes,
-                }}
-              >
-                <Ballot />
-              </BallotContext.Provider>
-            </Gamepad>
+            <WrongPrecinctScreen
+              useEffectToggleLargeDisplay={useEffectToggleLargeDisplay}
+            />
           )
         }
+        return (
+          <Gamepad onButtonDown={handleGamepadButtonDown}>
+            <BallotContext.Provider
+              value={{
+                machineConfig,
+                ballotStyleId,
+                contests,
+                electionDefinition: optionalElectionDefinition,
+                updateTally,
+                isLiveMode,
+                markVoterCardPrinted,
+                markVoterCardVoided,
+                precinctId,
+                printer,
+                resetBallot,
+                setUserSettings,
+                updateVote,
+                forceSaveVote,
+                userSettings,
+                votes: votes ?? blankBallotVotes,
+              }}
+            >
+              <Ballot />
+            </BallotContext.Provider>
+          </Gamepad>
+        )
       }
+    }
 
-      // font size may have been changed by previous voter, reset to defaults
-      this.setDocumentFontSize()
+    // font size may have been changed by previous voter, reset to defaults
+    setDocumentFontSize()
```

Comment:
Shouldn't this be in a `useEffect` callback?

Reply:
Turns out the test checking this passes when removed after this refactor!